vectordatasource/transform.py

```diff
@@ -3943,15 +3996,29 @@ def merge_networks_from_tags(shape, props, fid, zoom):
     ref = props.get('ref')
     mz_networks = props.get('mz_networks', [])
 
+    # if there's no network, but the operator indicates a network, then we can
+    # back-fill an approximate network tag from the operator. this can mean
+    # that extra refs are available for road networks.
+    if network is None:
+        operator = props.get('operator')
+        backfill_network = _NETWORK_OPERATORS.get(operator)
```

Comment:
This may need to get slightly more sophisticated so account for multiple road shield types per country also based on the A, N, M, & etc shield text prefixes &/or the road class?

Reply:
I think this needs to get a _lot_ more sophisticated!

The addition of the `operator`-based fall-back was only because the M1 section that #1062 mentioned had a `ref` but no `network` on the way, and was only part of the E13 relation (which has both `ref` and `network`). The current logic is built around tuples of `(route_type, network, ref)`, so the simplest approach was to guess a `network` based on the information available on the way.

This is a kludge, but it's an improvement over what went before.

Once we have #135, then we should change the logic so that we're additionally able to match on the country and road class. For example, a road in the UK with `highway=motorway` would be generally expected to have a `ref` starting with `M` (apart from [the ones which don't](https://en.wikipedia.org/wiki/A329(M)_motorway), which have been left scattered around like rakes in long grass by the trickster spirits to frustrate us), and could be assigned a `network` more accurately, where one wasn't otherwise specified.